tauri-app/src/lib/components/tables/OrderVaultsVolTable.svelte

```diff
@@ -0,0 +1,59 @@
+<script lang="ts">
+  import { createInfiniteQuery } from '@tanstack/svelte-query';
+  import TanstackAppTable from './TanstackAppTable.svelte';
+  import { QKEY_VAULTS_VOL_LIST } from '$lib/queries/keys';
+  import { orderVaultsVolume } from '$lib/queries/orderTradesList';
+  import { subgraphUrl } from '$lib/stores/settings';
+  import { TableBodyCell, TableHeadCell } from 'flowbite-svelte';
+  import Hash from '$lib/components/Hash.svelte';
+  import { HashType } from '$lib/types/hash';
+  import { formatUnits } from 'viem';
+  import TableTimeFilters from '../charts/TableTimeFilters.svelte';
+
+  let startTimestamp: number | undefined;
+  let endTimestamp: number | undefined;
+
+  export let id: string;
+
+  $: vaultsVol = createInfiniteQuery({
+    queryKey: [id, QKEY_VAULTS_VOL_LIST + id],
+    queryFn: () => orderVaultsVolume(id, $subgraphUrl || '', startTimestamp, endTimestamp),
+    initialPageParam: 0,
+    getNextPageParam: () => undefined,
+    enabled: !!$subgraphUrl,
+  });
+</script>
+
+<TanstackAppTable query={vaultsVol} emptyMessage="No trades found" rowHoverable={false}>
+  <svelte:fragment slot="timeFilter">
+    <TableTimeFilters bind:startTimestamp bind:endTimestamp />
+  </svelte:fragment>
+  <svelte:fragment slot="head">
+    <TableHeadCell padding="p-4">Vault</TableHeadCell>
+    <TableHeadCell padding="p-0">Token</TableHeadCell>
+    <TableHeadCell padding="p-0">In Volume</TableHeadCell>
+    <TableHeadCell padding="p-0">Out Volume</TableHeadCell>
+    <TableHeadCell padding="p-0">Total Volume</TableHeadCell>
+  </svelte:fragment>
+
+  <svelte:fragment slot="bodyRow" let:item>
+    <TableBodyCell tdClass="px-4 py-2">
+      <Hash type={HashType.Identifier} shorten value={item.id} />
+    </TableBodyCell>
+    <TableBodyCell tdClass="break-all py-2 min-w-32">
+      <div class="flex gap-x-3">
+        <Hash type={HashType.Identifier} shorten value={item.token.address} />
```

Comment:
The address isn't a hash, would be good if it showed another icon here instead of the fingerprint - do we need to add a new HashType?

Reply:
fixed